### PR TITLE
Upgrade to ErrorProne 2.4.0 (and others) to avoid JUnit 3.13-SNAPSHOT dependency

### PR DESCRIPTION
**Pull request by ejona86**
Added requireUpperBoundDeps to verify new versions. Upgraded Guava and JUnit to
avoid downgrading transitive dependencies. Upgraded auto-service as otherwise
javac crashed with "java.lang.ClassCastException: com.sun.tools.javac.util.List
cannot be cast to javax.lang.model.type.DeclaredType".

### CODE REVIEW

**Comment by ejona86** [comment]
CI will fail until #22 is merged. Afterward CI should pass. I will not merge until CI passes.

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**
